mars/deploy/oscar/session.py

```diff
@@ -1185,13 +1199,29 @@ async def fetch_infos(self, *tileables, fields, **kwargs) -> list:
                 for chunk in fetch_tileable.chunks:
                     chunks.append(chunk)
                     get_chunk_metas.append(
-                        self._meta_api.get_chunk_meta.delay(chunk.key, fields=["bands"])
+                        self._meta_api.get_chunk_meta.delay(
+                            chunk.key, fields=get_chunk_meta_fields
+                        )
                     )
                     fetch_infos.append(
                         ChunkFetchInfo(tileable=tileable, chunk=chunk, indexes=None)
                     )
                 fetch_infos_list.append(fetch_infos)
             chunk_metas = await self._meta_api.get_chunk_meta.batch(*get_chunk_metas)
+            if not _need_query_storage:
```

Comment:
Is it possible to put the logic without querying storage into a separate function? Now the `if not _need_query_storage` logic looks a bit messy and we can make the logic more clean.